### ISSUE TITLE
ArgumentError - unknown keywords: path, domain, expire_after, secure, httponly, defer, renew, sidbits, secure_random, secret, coder:

### ISSUE THREAD

**Issue by gingerlime** [redis/redis-rb#921]
After upgrading to redis 4.1.4 to 4.2.1 we get this error when accessing the Sidekiq web dashboard

`ArgumentError - unknown keywords: path, domain, expire_after, secure, httponly, defer, renew, sidbits, secure_random, secret, coder:`

full (sanitized) trace

```
web_1          | 2020-06-15 10:03:28.614623 FATAL name=Rails message= exception=[ArgumentErro
r: unknown keywords: path, domain, expire_after, secure, httponly, defer, renew, sidbits, secure_random, secret, coder: /usr/local/bundle/gems/redis-4.2.1/lib/redis.rb:836:in `set
'
web_1          | /usr/local/bundle/gems/redis-store-1.8.2/lib/redis/store/interface.rb:9:in `set'
web_1          | /usr/local/bundle/gems/redis-store-1.8.2/lib/redis/store/ttl.rb:8:in `set'
web_1          | /usr/local/bundle/gems/redis-store-1.8.2/lib/redis/store/serialization.rb:5:in `block in set'
web_1          | /usr/local/bundle/gems/redis-store-1.8.2/lib/redis/store/serialization.rb:40:in `_marshal'
web_1          | /usr/local/bundle/gems/redis-store-1.8.2/lib/redis/store/serialization.rb:5:in `set'
web_1          | /usr/local/bundle/gems/redis-store-1.8.2/lib/redis/store/namespace.rb:7:in `block in set'
web_1          | /usr/local/bundle/gems/redis-store-1.8.2/lib/redis/store/namespace.rb:144:in `namespace'
web_1          | /usr/local/bundle/gems/redis-store-1.8.2/lib/redis/store/namespace.rb:7:in `set'
web_1          | /usr/local/bundle/gems/redis-rack-2.1.0/lib/rack/session/redis.rb:49:in `block (2 levels) in write_session'
web_1          | /usr/local/bundle/gems/redis-rack-2.1.0/lib/redis/rack/connection.rb:22:in `with'
web_1          | /usr/local/bundle/gems/redis-rack-2.1.0/lib/rack/session/redis.rb:82:in `with'
web_1          | /usr/local/bundle/gems/redis-rack-2.1.0/lib/rack/session/redis.rb:49:in `block in write_session'
web_1          | /usr/local/bundle/gems/redis-rack-2.1.0/lib/rack/session/redis.rb:70:in `with_lock'
web_1          | /usr/local/bundle/gems/redis-rack-2.1.0/lib/rack/session/redis.rb:48:in `write_session'
web_1          | /usr/local/bundle/gems/rack-2.2.2/lib/rack/session/abstract/id.rb:388:in `commit_session'
web_1          | /usr/local/bundle/gems/rack-2.2.2/lib/rack/session/abstract/id.rb:268:in `context'
web_1          | /usr/local/bundle/gems/rack-2.2.2/lib/rack/session/abstract/id.rb:260:in `call'
web_1          | /usr/local/bundle/gems/actionpack-6.0.3.1/lib/action_dispatch/middleware/cookies.rb:648:in `call'
web_1          | /usr/local/bundle/gems/activerecord-6.0.3.1/lib/active_record/migration.rb:567:in `call'
web_1          | /usr/local/bundle/gems/actionpack-6.0.3.1/lib/action_dispatch/middleware/callbacks.rb:27:in `block in call'
web_1          | /usr/local/bundle/gems/activesupport-6.0.3.1/lib/active_support/callbacks.rb:101:in `run_callbacks'
web_1          | /usr/local/bundle/gems/actionpack-6.0.3.1/lib/action_dispatch/middleware/callbacks.rb:26:in `call'
web_1          | /usr/local/bundle/gems/actionpack-6.0.3.1/lib/action_dispatch/middleware/executor.rb:14:in `call'
web_1          | /usr/local/bundle/gems/actionpack-6.0.3.1/lib/action_dispatch/middleware/actionable_exceptions.rb:17:in `call'
web_1          | /usr/local/bundle/gems/rollbar-2.25.1/lib/rollbar/middleware/rails/rollbar.rb:25:in `block in call'
web_1          | /usr/local/bundle/gems/rollbar-2.25.1/lib/rollbar.rb:145:in `scoped'
web_1          | /usr/local/bundle/gems/rollbar-2.25.1/lib/rollbar/middleware/rails/rollbar.rb:22:in `call'
web_1          | /usr/local/bundle/gems/actionpack-6.0.3.1/lib/action_dispatch/middleware/debug_exceptions.rb:32:in `call'
web_1          | /usr/local/bundle/gems/rollbar-2.25.1/lib/rollbar/middleware/rails/show_exceptions.rb:22:in `call_with_rollbar'
web_1          | /usr/local/bundle/gems/ddtrace-0.35.2/lib/ddtrace/contrib/rails/middlewares.rb:17:in `call'
web_1          | /usr/local/bundle/gems/actionpack-6.0.3.1/lib/action_dispatch/middleware/show_exceptions.rb:33:in `call'
web_1          | /usr/local/bundle/gems/rails_semantic_logger-4.4.4/lib/rails_semantic_logger/rack/logger.rb:43:in `call_app'
web_1          | /usr/local/bundle/gems/rails_semantic_logger-4.4.4/lib/rails_semantic_logger/rack/logger.rb:26:in `block in call'
web_1          | /usr/local/bundle/gems/semantic_logger-4.6.1/lib/semantic_logger/semantic_logger.rb:386:in `named_tagged'
web_1          | /usr/local/bundle/gems/semantic_logger-4.6.1/lib/semantic_logger/base.rb:194:in `tagged'
web_1          | /usr/local/bundle/gems/rails_semantic_logger-4.4.4/lib/rails_semantic_logger/rack/logger.rb:26:in `call'
web_1          | /usr/local/bundle/gems/actionpack-6.0.3.1/lib/action_dispatch/middleware/remote_ip.rb:81:in `call'
web_1          | /usr/local/bundle/gems/request_store-1.4.1/lib/request_store/middleware.rb:19:in `call'
web_1          | /usr/local/bundle/gems/actionpack-6.0.3.1/lib/action_dispatch/middleware/request_id.rb:27:in `call'
web_1          | /usr/local/bundle/gems/rack-2.2.2/lib/rack/method_override.rb:24:in `call'
web_1          | /usr/local/bundle/gems/rack-2.2.2/lib/rack/runtime.rb:22:in `call'
web_1          | /usr/local/bundle/gems/activesupport-6.0.3.1/lib/active_support/cache/strategy/local_cache_middleware.rb:29:in `call'
web_1          | /usr/local/bundle/gems/actionpack-6.0.3.1/lib/action_dispatch/middleware/executor.rb:14:in `call'
web_1          | /usr/local/bundle/gems/actionpack-6.0.3.1/lib/action_dispatch/middleware/static.rb:126:in `call'
web_1          | /usr/local/bundle/gems/rack-2.2.2/lib/rack/sendfile.rb:110:in `call'
web_1          | /usr/local/bundle/gems/actionpack-6.0.3.1/lib/action_dispatch/middleware/host_authorization.rb:82:in `call'
web_1          | /usr/local/bundle/gems/rack-mini-profiler-2.0.2/lib/mini_profiler/profiler.rb:200:in `call'
web_1          | /usr/local/bundle/gems/ddtrace-0.35.2/lib/ddtrace/contrib/rack/middlewares.rb:85:in `call'
web_1          | /usr/local/bundle/gems/webpacker-5.1.1/lib/webpacker/dev_server_proxy.rb:25:in `perform_request'
web_1          | /usr/local/bundle/gems/rack-proxy-0.6.5/lib/rack/proxy.rb:57:in `call'
web_1          | /usr/local/bundle/gems/railties-6.0.3.1/lib/rails/engine.rb:527:in `call'
web_1          | /usr/local/bundle/gems/railties-6.0.3.1/lib/rails/railtie.rb:190:in `public_send'
web_1          | /usr/local/bundle/gems/railties-6.0.3.1/lib/rails/railtie.rb:190:in `method_missing'
web_1          | /usr/local/bundle/gems/puma-3.12.4/lib/puma/configuration.rb:227:in `call'
web_1          | /usr/local/bundle/gems/puma-3.12.4/lib/puma/server.rb:675:in `handle_request'
web_1          | /usr/local/bundle/gems/puma-3.12.4/lib/puma/server.rb:476:in `process_client'
web_1          | /usr/local/bundle/gems/puma-3.12.4/lib/puma/server.rb:334:in `block in run'
web_1          | /usr/local/bundle/gems/puma-3.12.4/lib/puma/thread_pool.rb:135:in `block in spawn_thread']
```

is this a problem with sidekiq?

**Comment by byroot** [comment]
Hi.

No this is a compatibility issue between redis-rb 4.2 and `redis-store`. This was already reported, and was fixed in `redis-store` (https://github.com/redis-store/redis-store/pull/333).

A new `redis-store` will likely be published soon, so I'd recommend to simply lock redis-rb to 4.1 in the meantime.

**Comment by gingerlime** [comment]
Thanks @byroot. I really appreciate it. I checked redis-store but didn't see any open issues... should have checked the closed ones.

**Comment by Paul-Yves** [comment]
Now that the issue is fixed, maybe redis should require the 1.9 version of redis-store, my gemfile.lock still specify `redis-store (>= 1.2, < 2)` which is no good since it attempts to download redis-store (1.8.2)
